### PR TITLE
[palsigslot] New port

### LINKED ISSUE #15557: [New Port Request] palsigslot
Library creator: [Pierre-Antoine Lacaze](https://github.com/palacaze)

Library name: palsigslot

Library description: A header-only, thread safe implementation of signal-slots for C++.

Source repository URL: https://github.com/palacaze/sigslot

### PR DESCRIPTION
**Pull request by luca-schlecker**
Signed-off-by: Luca Schlecker <luca.schlecker@hotmail.com>

- What does your PR fix? Fixes #15557

- Which triplets are supported/not supported? All CI triplets compiling ✔️

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes ✔️

### CODE REVIEW

File: ports/palsigslot/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "palsigslot",
+  "version-string": "1.2.0",
+  "description": "A header-only, thread safe implementation of signal-slots for C++."
```
Comment (NancyLi1013) [inline ports/palsigslot/vcpkg.json:4]:
```suggestion
  "description": "A header-only, thread safe implementation of signal-slots for C++.",
  "homepage": "https://github.com/palacaze/sigslot"
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @luca-schlecker.

File: ports/palsigslot/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    SHA512 6a6fb862a9eeea78732f2191916c7384a4bcb65e42c948f726459ee3cb446f90c4bd18c7c94ee4f9b15c81c5aa729094013805161d532c2284d9e77cdffaa468
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+vcpkg_install_cmake()
```
Comment (dan-shaw) [inline ports/palsigslot/portfile.cmake:14]: Since this is a header only library, can we copy the header files instead of CMake configure and install?

Comment (luca-schlecker) [reply ports/palsigslot/portfile.cmake:14]: During the install step needed CMake files for using `find_package(PalSigslot)` are generated. Leaving out this step these files would either have to be left out and `find_path(PALSIGSLOT_INCLUDE_DIRS sigslot/signal.hpp)` had to be used or they'd have to be copied over manually with them stored inside the port directory. I prefer installing it, but if desired, I can update this PR to leave out the install step.

Comment (luca-schlecker) [reply ports/palsigslot/portfile.cmake:17]: To help compile-time I used options given by `PalSigslot` to leave out test and example compilation 👍

**Pull request merged**